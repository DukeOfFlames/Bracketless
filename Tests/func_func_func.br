START
pylib functools
pylib math
Â°combined_func = {functools.reduce}{fn {f, g} { -> f . g }, {math.gamma, math.sqrt, math.log2}}
#drucke{combined_func}
#drucke{combined_func{562949953421312}}
END