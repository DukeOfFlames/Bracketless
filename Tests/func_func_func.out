InterpreterNode.Function
  {
    type:
      FunctionType.Internal
    body:
      Python Function:
        Name:
          call
        Captured Variables:
          {
            python_func:
              Python Function:
                Name:
                  call
                Captured Variables:
                  {
                    bracketless_func:
                      InterpreterNode.Function
                        {
                          type:
                            FunctionType.Internal
                          body:
                            Python Function:
                              Name:
                                combined_func
                              Captured Variables:
                                {
                                  lhs:
                                    InterpreterNode.Function
                                      {
                                        type:
                                          FunctionType.Internal
                                        body:
                                          Python Function:
                                            Name:
                                              call
                                            Captured Variables:
                                              {
                                                python_func:
                                                  Python Function:
                                                    Name:
                                                      call
                                                    Captured Variables:
                                                      {
                                                        bracketless_func:
                                                          InterpreterNode.Function
                                                            {
                                                              type:
                                                                FunctionType.Internal
                                                              body:
                                                                Python Function:
                                                                  Name:
                                                                    combined_func
                                                                  Captured Variables:
                                                                    {
                                                                      lhs:
                                                                        InterpreterNode.Function
                                                                          {
                                                                            type:
                                                                              FunctionType.Internal
                                                                            body:
                                                                              Python Function:
                                                                                Name:
                                                                                  call
                                                                                Captured Variables:
                                                                                  {
                                                                                    python_func:
                                                                                      Python Function:
                                                                                        Name:
                                                                                          call
                                                                                        Captured Variables:
                                                                                          {
                                                                                            bracketless_func:
                                                                                              InterpreterNode.Function
                                                                                                {
                                                                                                  type:
                                                                                                    FunctionType.Internal
                                                                                                  body:
                                                                                                    Python Function:
                                                                                                      Name:
                                                                                                        call
                                                                                                      Captured Variables:
                                                                                                        {
                                                                                                          python_func:
                                                                                                            Python Function:
                                                                                                              Name:
                                                                                                                gamma
                                                                                                              Code:
                                                                                                                <builtin>
                                                                                                        }
                                                                                                      Code:
                                                                                                        LOAD_GLOBAL
                                                                                                        LOAD_DEREF
                                                                                                        LOAD_CONST
                                                                                                        LOAD_CONST
                                                                                                        MAKE_FUNCTION
                                                                                                        LOAD_FAST
                                                                                                        GET_ITER
                                                                                                        CALL_FUNCTION
                                                                                                        CALL_FUNCTION_EX
                                                                                                        CALL_FUNCTION
                                                                                                        RETURN_VALUE
                                                                                                }
                                                                                          }
                                                                                        Code:
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_ATTR
                                                                                          LOAD_ATTR
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_ATTR
                                                                                          LOAD_ATTR
                                                                                          LOAD_DEREF
                                                                                          CALL_FUNCTION
                                                                                          CALL_FUNCTION
                                                                                          LOAD_CONST
                                                                                          LOAD_CONST
                                                                                          MAKE_FUNCTION
                                                                                          LOAD_FAST
                                                                                          GET_ITER
                                                                                          CALL_FUNCTION
                                                                                          LOAD_CONST
                                                                                          BUILD_CONST_KEY_MAP
                                                                                          CALL_FUNCTION
                                                                                          LOAD_METHOD
                                                                                          LOAD_CONST
                                                                                          CALL_METHOD
                                                                                          CALL_FUNCTION
                                                                                          RETURN_VALUE
                                                                                  }
                                                                                Code:
                                                                                  LOAD_GLOBAL
                                                                                  LOAD_DEREF
                                                                                  LOAD_CONST
                                                                                  LOAD_CONST
                                                                                  MAKE_FUNCTION
                                                                                  LOAD_FAST
                                                                                  GET_ITER
                                                                                  CALL_FUNCTION
                                                                                  CALL_FUNCTION_EX
                                                                                  CALL_FUNCTION
                                                                                  RETURN_VALUE
                                                                          }
                                                                      rhs:
                                                                        InterpreterNode.Function
                                                                          {
                                                                            type:
                                                                              FunctionType.Internal
                                                                            body:
                                                                              Python Function:
                                                                                Name:
                                                                                  call
                                                                                Captured Variables:
                                                                                  {
                                                                                    python_func:
                                                                                      Python Function:
                                                                                        Name:
                                                                                          call
                                                                                        Captured Variables:
                                                                                          {
                                                                                            bracketless_func:
                                                                                              InterpreterNode.Function
                                                                                                {
                                                                                                  type:
                                                                                                    FunctionType.Internal
                                                                                                  body:
                                                                                                    Python Function:
                                                                                                      Name:
                                                                                                        call
                                                                                                      Captured Variables:
                                                                                                        {
                                                                                                          python_func:
                                                                                                            Python Function:
                                                                                                              Name:
                                                                                                                sqrt
                                                                                                              Code:
                                                                                                                <builtin>
                                                                                                        }
                                                                                                      Code:
                                                                                                        LOAD_GLOBAL
                                                                                                        LOAD_DEREF
                                                                                                        LOAD_CONST
                                                                                                        LOAD_CONST
                                                                                                        MAKE_FUNCTION
                                                                                                        LOAD_FAST
                                                                                                        GET_ITER
                                                                                                        CALL_FUNCTION
                                                                                                        CALL_FUNCTION_EX
                                                                                                        CALL_FUNCTION
                                                                                                        RETURN_VALUE
                                                                                                }
                                                                                          }
                                                                                        Code:
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_ATTR
                                                                                          LOAD_ATTR
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_GLOBAL
                                                                                          LOAD_ATTR
                                                                                          LOAD_ATTR
                                                                                          LOAD_DEREF
                                                                                          CALL_FUNCTION
                                                                                          CALL_FUNCTION
                                                                                          LOAD_CONST
                                                                                          LOAD_CONST
                                                                                          MAKE_FUNCTION
                                                                                          LOAD_FAST
                                                                                          GET_ITER
                                                                                          CALL_FUNCTION
                                                                                          LOAD_CONST
                                                                                          BUILD_CONST_KEY_MAP
                                                                                          CALL_FUNCTION
                                                                                          LOAD_METHOD
                                                                                          LOAD_CONST
                                                                                          CALL_METHOD
                                                                                          CALL_FUNCTION
                                                                                          RETURN_VALUE
                                                                                  }
                                                                                Code:
                                                                                  LOAD_GLOBAL
                                                                                  LOAD_DEREF
                                                                                  LOAD_CONST
                                                                                  LOAD_CONST
                                                                                  MAKE_FUNCTION
                                                                                  LOAD_FAST
                                                                                  GET_ITER
                                                                                  CALL_FUNCTION
                                                                                  CALL_FUNCTION_EX
                                                                                  CALL_FUNCTION
                                                                                  RETURN_VALUE
                                                                          }
                                                                    }
                                                                  Code:
                                                                    LOAD_GLOBAL
                                                                    LOAD_FAST
                                                                    CALL_FUNCTION
                                                                    LOAD_CONST
                                                                    COMPARE_OP
                                                                    POP_JUMP_IF_FALSE
                                                                    LOAD_GLOBAL
                                                                    RAISE_VARARGS
                                                                    LOAD_FAST
                                                                    LOAD_CONST
                                                                    BINARY_SUBSCR
                                                                    STORE_FAST
                                                                    LOAD_GLOBAL
                                                                    LOAD_GLOBAL
                                                                    LOAD_ATTR
                                                                    LOAD_ATTR
                                                                    LOAD_GLOBAL
                                                                    LOAD_GLOBAL
                                                                    LOAD_GLOBAL
                                                                    LOAD_ATTR
                                                                    LOAD_ATTR
                                                                    LOAD_DEREF
                                                                    CALL_FUNCTION
                                                                    CALL_FUNCTION
                                                                    LOAD_GLOBAL
                                                                    LOAD_GLOBAL
                                                                    LOAD_ATTR
                                                                    LOAD_ATTR
                                                                    LOAD_GLOBAL
                                                                    LOAD_GLOBAL
                                                                    LOAD_GLOBAL
                                                                    LOAD_ATTR
                                                                    LOAD_ATTR
                                                                    LOAD_DEREF
                                                                    CALL_FUNCTION
                                                                    CALL_FUNCTION
                                                                    LOAD_GLOBAL
                                                                    LOAD_GLOBAL
                                                                    LOAD_ATTR
                                                                    LOAD_ATTR
                                                                    LOAD_FAST
                                                                    CALL_FUNCTION
                                                                    BUILD_LIST
                                                                    LOAD_CONST
                                                                    BUILD_CONST_KEY_MAP
                                                                    CALL_FUNCTION
                                                                    BUILD_LIST
                                                                    LOAD_CONST
                                                                    BUILD_CONST_KEY_MAP
                                                                    CALL_FUNCTION
                                                                    LOAD_METHOD
                                                                    LOAD_CONST
                                                                    CALL_METHOD
                                                                    RETURN_VALUE
                                                            }
                                                      }
                                                    Code:
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_ATTR
                                                      LOAD_ATTR
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_ATTR
                                                      LOAD_ATTR
                                                      LOAD_DEREF
                                                      CALL_FUNCTION
                                                      CALL_FUNCTION
                                                      LOAD_CONST
                                                      LOAD_CONST
                                                      MAKE_FUNCTION
                                                      LOAD_FAST
                                                      GET_ITER
                                                      CALL_FUNCTION
                                                      LOAD_CONST
                                                      BUILD_CONST_KEY_MAP
                                                      CALL_FUNCTION
                                                      LOAD_METHOD
                                                      LOAD_CONST
                                                      CALL_METHOD
                                                      CALL_FUNCTION
                                                      RETURN_VALUE
                                              }
                                            Code:
                                              LOAD_GLOBAL
                                              LOAD_DEREF
                                              LOAD_CONST
                                              LOAD_CONST
                                              MAKE_FUNCTION
                                              LOAD_FAST
                                              GET_ITER
                                              CALL_FUNCTION
                                              CALL_FUNCTION_EX
                                              CALL_FUNCTION
                                              RETURN_VALUE
                                      }
                                  rhs:
                                    InterpreterNode.Function
                                      {
                                        type:
                                          FunctionType.Internal
                                        body:
                                          Python Function:
                                            Name:
                                              call
                                            Captured Variables:
                                              {
                                                python_func:
                                                  Python Function:
                                                    Name:
                                                      call
                                                    Captured Variables:
                                                      {
                                                        bracketless_func:
                                                          InterpreterNode.Function
                                                            {
                                                              type:
                                                                FunctionType.Internal
                                                              body:
                                                                Python Function:
                                                                  Name:
                                                                    call
                                                                  Captured Variables:
                                                                    {
                                                                      python_func:
                                                                        Python Function:
                                                                          Name:
                                                                            log2
                                                                          Code:
                                                                            <builtin>
                                                                    }
                                                                  Code:
                                                                    LOAD_GLOBAL
                                                                    LOAD_DEREF
                                                                    LOAD_CONST
                                                                    LOAD_CONST
                                                                    MAKE_FUNCTION
                                                                    LOAD_FAST
                                                                    GET_ITER
                                                                    CALL_FUNCTION
                                                                    CALL_FUNCTION_EX
                                                                    CALL_FUNCTION
                                                                    RETURN_VALUE
                                                            }
                                                      }
                                                    Code:
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_ATTR
                                                      LOAD_ATTR
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_GLOBAL
                                                      LOAD_ATTR
                                                      LOAD_ATTR
                                                      LOAD_DEREF
                                                      CALL_FUNCTION
                                                      CALL_FUNCTION
                                                      LOAD_CONST
                                                      LOAD_CONST
                                                      MAKE_FUNCTION
                                                      LOAD_FAST
                                                      GET_ITER
                                                      CALL_FUNCTION
                                                      LOAD_CONST
                                                      BUILD_CONST_KEY_MAP
                                                      CALL_FUNCTION
                                                      LOAD_METHOD
                                                      LOAD_CONST
                                                      CALL_METHOD
                                                      CALL_FUNCTION
                                                      RETURN_VALUE
                                              }
                                            Code:
                                              LOAD_GLOBAL
                                              LOAD_DEREF
                                              LOAD_CONST
                                              LOAD_CONST
                                              MAKE_FUNCTION
                                              LOAD_FAST
                                              GET_ITER
                                              CALL_FUNCTION
                                              CALL_FUNCTION_EX
                                              CALL_FUNCTION
                                              RETURN_VALUE
                                      }
                                }
                              Code:
                                LOAD_GLOBAL
                                LOAD_FAST
                                CALL_FUNCTION
                                LOAD_CONST
                                COMPARE_OP
                                POP_JUMP_IF_FALSE
                                LOAD_GLOBAL
                                RAISE_VARARGS
                                LOAD_FAST
                                LOAD_CONST
                                BINARY_SUBSCR
                                STORE_FAST
                                LOAD_GLOBAL
                                LOAD_GLOBAL
                                LOAD_ATTR
                                LOAD_ATTR
                                LOAD_GLOBAL
                                LOAD_GLOBAL
                                LOAD_GLOBAL
                                LOAD_ATTR
                                LOAD_ATTR
                                LOAD_DEREF
                                CALL_FUNCTION
                                CALL_FUNCTION
                                LOAD_GLOBAL
                                LOAD_GLOBAL
                                LOAD_ATTR
                                LOAD_ATTR
                                LOAD_GLOBAL
                                LOAD_GLOBAL
                                LOAD_GLOBAL
                                LOAD_ATTR
                                LOAD_ATTR
                                LOAD_DEREF
                                CALL_FUNCTION
                                CALL_FUNCTION
                                LOAD_GLOBAL
                                LOAD_GLOBAL
                                LOAD_ATTR
                                LOAD_ATTR
                                LOAD_FAST
                                CALL_FUNCTION
                                BUILD_LIST
                                LOAD_CONST
                                BUILD_CONST_KEY_MAP
                                CALL_FUNCTION
                                BUILD_LIST
                                LOAD_CONST
                                BUILD_CONST_KEY_MAP
                                CALL_FUNCTION
                                LOAD_METHOD
                                LOAD_CONST
                                CALL_METHOD
                                RETURN_VALUE
                        }
                  }
                Code:
                  LOAD_GLOBAL
                  LOAD_GLOBAL
                  LOAD_GLOBAL
                  LOAD_ATTR
                  LOAD_ATTR
                  LOAD_GLOBAL
                  LOAD_GLOBAL
                  LOAD_GLOBAL
                  LOAD_ATTR
                  LOAD_ATTR
                  LOAD_DEREF
                  CALL_FUNCTION
                  CALL_FUNCTION
                  LOAD_CONST
                  LOAD_CONST
                  MAKE_FUNCTION
                  LOAD_FAST
                  GET_ITER
                  CALL_FUNCTION
                  LOAD_CONST
                  BUILD_CONST_KEY_MAP
                  CALL_FUNCTION
                  LOAD_METHOD
                  LOAD_CONST
                  CALL_METHOD
                  CALL_FUNCTION
                  RETURN_VALUE
          }
        Code:
          LOAD_GLOBAL
          LOAD_DEREF
          LOAD_CONST
          LOAD_CONST
          MAKE_FUNCTION
          LOAD_FAST
          GET_ITER
          CALL_FUNCTION
          CALL_FUNCTION_EX
          CALL_FUNCTION
          RETURN_VALUE
  }
720.0
